### PR TITLE
calculate gas margin 2000 (20%)

### PR DESCRIPTION
**Pull request by chef-plankton**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

File: apps/web/src/state/multicall/updater.tsx [hunk 1]
```diff
@@ -46,6 +46,8 @@ async function fetchChunk(
         blockTag: minBlockNumber,
       }
     )
+    console.log(resultsBlockNumber);
```
Comment (simone1999) [inline apps/web/src/state/multicall/updater.tsx]: is this needed?

File: apps/web/src/utils/exchange.ts [hunk 2]
```diff
@@ -39,7 +40,7 @@ export function useRouterContract() {
 }
 
 export function useAkkaRouterContract() {
-  return useContract<AkkaRouter>(ROUTER_ADDRESS[ChainId.BITGERT].Akka, AKKA_BTGERT_ABI, true)
+  return useContract<AkkaRouter>(ROUTER_ADDRESS_COMMON_AKKA, AKKA_BTGERT_ABI, true)
```
Comment (simone1999) [inline apps/web/src/utils/exchange.ts]: please get the active chain ID here and use the correct address for the chain, this way you cn desply your router also on different addresses and upgrade it.

File: apps/web/src/views/Swap/AkkaSwap/hooks/useApproveCallbackFromAkkaTrade.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
+import { ROUTER_ADDRESS_COMMON_AKKA } from 'config/constants/exchange'
 import { ChainId, Currency, CurrencyAmount } from '@pancakeswap/sdk'
 import { useApproveCallback } from 'hooks/useApproveCallback'
 import { useMemo } from 'react'
 
 // wraps useApproveCallback in the context of a swap
 export function useApproveCallbackFromAkkaTrade(inputAmount: CurrencyAmount<Currency>) {
   const amountToApprove = useMemo(() => inputAmount || undefined, [inputAmount])
-  return useApproveCallback(amountToApprove, ROUTER_ADDRESS[ChainId.BITGERT].Akka)
+  return useApproveCallback(amountToApprove, ROUTER_ADDRESS_COMMON_AKKA)
```
Comment (simone1999) [inline apps/web/src/views/Swap/AkkaSwap/hooks/useApproveCallbackFromAkkaTrade.ts]: also here, please stay with ROUTER_ADDRESS and use the current chainId

File: apps/web/src/views/Swap/index.tsx [hunk 4]
```diff
@@ -69,10 +69,10 @@ export default function Swap() {
 
   // Check if pancakeswap route is better than akka route or not
   useEffect(() => {
-    if (akkaRouterTrade?.route?.returnAmount && v2Trade?.outputAmount?.toSignificant(6)) {
-      if (Number(v2Trade?.outputAmount?.toSignificant(6)) > Number(akkaRouterTrade?.route?.returnAmount)) {
+    if (akkaRouterTrade?.route?.returnAmountWei && v2Trade?.outputAmount) {
+      if (v2Trade?.outputAmount.numerator.toString() > akkaRouterTrade?.route?.returnAmountWei) {
```
Comment (simone1999) [inline apps/web/src/views/Swap/index.tsx]: looks like a comparision between string and number

File: apps/web/src/views/Swap/index.tsx [hunk 5]
```diff
@@ -69,10 +69,10 @@ export default function Swap() {
...
         toggleSetAkkaModeToFalse()
-      } else if (Number(v2Trade?.outputAmount?.toSignificant(6)) < Number(akkaRouterTrade?.route?.returnAmount)) {
+      } else if (v2Trade?.outputAmount.numerator.toString() < akkaRouterTrade?.route?.returnAmountWei) {
```
Comment (simone1999) [inline apps/web/src/views/Swap/index.tsx]: again comparing string and number?

**Review by simone1999 — CHANGES_REQUESTED** [review]

File: apps/web/src/state/multicall/updater.tsx (hunk 1 above)
Comment (chef-plankton) [reply apps/web/src/state/multicall/updater.tsx:49]: No I will delete it

Comment (chef-plankton) [reply apps/web/src/state/multicall/updater.tsx]: > is this needed?

No I will delete it

File: apps/web/src/utils/exchange.ts (hunk 2 above)
Comment (chef-plankton) [reply apps/web/src/utils/exchange.ts]: > please get the active chain ID here and use the correct address for the chain, this way you cn desply your router also on different addresses and upgrade it.

Ok sure

File: apps/web/src/views/Swap/AkkaSwap/hooks/useApproveCallbackFromAkkaTrade.ts (hunk 3 above)
Comment (chef-plankton) [reply apps/web/src/views/Swap/AkkaSwap/hooks/useApproveCallbackFromAkkaTrade.ts]: > also here, please stay with ROUTER_ADDRESS and use the current chainId

ok

File: apps/web/src/views/Swap/index.tsx (hunk 4 above)
Comment (chef-plankton) [reply apps/web/src/views/Swap/index.tsx]: > looks like a comparision between string and number

It's string now but we are going to compare both using bigNumber

File: apps/web/src/views/Swap/index.tsx (hunk 5 above)
Comment (chef-plankton) [reply apps/web/src/views/Swap/index.tsx]: > again comparing string and number?

It's string now but we are going to compare both using bigNumber

**Review by simone1999 — APPROVED** [review]

**Pull request merged**